WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/Bulk Edit Price/BulkUpdatePriceViewController.swift

```diff
@@ -0,0 +1,193 @@
+import Foundation
+import UIKit
+import Yosemite
+import Combine
+import WordPressKit
+
+// MARK: - BulkUpdatePriceViewController
+//
+final class BulkUpdatePriceViewController: UIViewController {
+
+    @IBOutlet weak private var tableView: UITableView!
+    @IBOutlet weak var saveButtonToBottom: NSLayoutConstraint!
+    @IBOutlet weak var saveButton: ButtonActivityIndicator!
+
+    private var viewModel: BulkUpdatePriceSettingsViewModel
+    private var subscriptions = Set<AnyCancellable>()
+
+    /// Tracking when the keyboard appears to keep the save button visible.
+    ///
+    private lazy var keyboardFrameObserver = KeyboardFrameObserver(onKeyboardFrameUpdate: { [weak self] frame in
+        self?.handleKeyboardFrameUpdate(keyboardFrame: frame)
+    })
+
+    init(viewModel: BulkUpdatePriceSettingsViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureTitleAndBackground()
+        configureTableView()
+        configureSaveButton()
+        configureViewModel()
+        keyboardFrameObserver.startObservingKeyboardFrame(sendInitialEvent: true)
+    }
+
+    /// Configures the  title and appearance of the save button
+    ///
+    func configureSaveButton() {
+        saveButton.applyPrimaryButtonStyle()
+        saveButton.setTitle(Localization.saveButtonTitle, for: .normal)
+        saveButton.addTarget(self, action: #selector(saveButtonTapped), for: .touchUpInside)
+    }
+
+    /// Configures the screen title and background
+    ///
+    private func configureTitleAndBackground() {
+        title = viewModel.screenTitle()
+        view.backgroundColor = .listBackground
+    }
+
+    /// Setup receiving updates for data changes from the view model
+    ///
+    private func configureViewModel() {
+        viewModel.$saveButtonState.sink { [weak self] state in
+            self?.saveButton.isEnabled = state == .enabled
+
+            if state == .loading {
+                self?.saveButton.showActivityIndicator()
+                // Dismiss the keyboard while we are in loading state
+                self?.view.endEditing(true)
```

Comment:
This instruction does not seem right here. Don't you think we should move it to `saveButtonTapped`?
Or maybe expose a different state for the keyboard handling 🤔

Reply:
I think It will be easier to follow & simpler if I added to the `saveButtonTapped()`, we will also have the extra space to display the validation errors if any (in landscape it left a little cramped). I updated the PR.